orc8r/cloud/go/services/bootstrapper/servicers/bootstrapper_test.go

```diff
@@ -244,11 +245,22 @@ func testWithGatewayBootstrapper(t *testing.T, networkId string) {
 	mdc.BootstrapConfig.ChallengeKey = dir + "/gw_challenge.key"
 	config.OverwriteMagmadConfigs(mdc)
 
+	// Create local directory for holding snowflake info.
+	// Ideally we could pass in a virtualized fs interface, but as of now,
+	// snowflake makes direct calls to os.
+	magmaRoot, ok := os.LookupEnv("MAGMA_ROOT")
+	assert.True(t, ok)
```

Comment:
This introduces a new dependency which won't allow to run this test on GW VMs/container/bare metal without MAGMA_ROOT env.
I am able to run this test on every gateway VM, I need without MAGMA_ROOT:

root@docker-desktop:/magma/orc8r/cloud/go/services/bootstrapper# echo $MAGMA_ROOT

root@docker-desktop:/magma/orc8r/cloud/go/services/bootstrapper# go test ./...
?   	magma/orc8r/cloud/go/services/bootstrapper	[no test files]
?   	magma/orc8r/cloud/go/services/bootstrapper/bootstrapper	[no test files]
ok  	magma/orc8r/cloud/go/services/bootstrapper/servicers	(cached)

This change will break this kind of portability of the test.

I see two options here:
1) Do not fail on unset MAGMA_ROOT, but use the default snowflake path (empty parameter list for Initialize())
2) Use default OS tmp dir/file (take a look at https://golang.org/pkg/os/#TempDir and/or https://golang.org/pkg/io/ioutil/#TempDir, https://golang.org/pkg/io/ioutil/#TempFile)

Reply:
Good catch, thanks for the input. I'm going to try out the tmpfile pattern, to keep things uniform / remove that MAGMA_ROOT dependency